### DISCUSSION TITLE
Doctrine/Dbal 3.0 compatiblity issue in a Symfony project

### DISCUSSION THREAD

**Discussion by hantsy** [doctrine/doctrine2#9164] [Support Questions] [answered]
I updated my [Symfony sample project](https://github.com/hantsy/symfony5-sample/tree/master/rest-sample) using Github dependenbot workflow, and keep updated with the latest stack.

In these days, I updated the dependencies and try to run the application by `symfony server:start` and hit the the `/posts` endpoint using `curl`, it throws exceptions like this in the terminal.

```bash
[Web Server ] [31-Oct-2021 03:34:21 UTC] [critical] Uncaught PHP Exception Doctrine\ORM\Exception\NotSupported: "Feature was deprecated in doctrine/dbal 2.x and is not supported by ins
talled doctrine/dbal:3.x, please see the doctrine/deprecations logs for new alternative approaches." at D:\hantsylabs\symfony5-sample\rest-sample\vendor\doctrine\orm\lib\Doctrine\ORM\Exception\NotSupported.php line 16
```

From the Symfony stack channel, the guys told me it was caused by the **UUID** generator, but from the exception message, I can not find any info about this , and also open the docs of `GeneratedValue` attribute,  no any info about the `UUID` is deprecated.

**Comment by SenseException** [comment, marked as answer]
There's an entry in the upgrade file that references the UUID for 3.0: https://github.com/doctrine/dbal/blob/cd95cf11dae73ae28c3084581810fbea057b36f2/UPGRADE.md#bc-break-removed-support-for-db-generated-uuids

BC breaks are possible in major releases, but I agree that this should be mentioned in the docs.

**Reply by derrabus** [reply]
Also, the exception message is indeed not helpful.

**Reply by hantsy** [reply]
Thanks for updating me the changelogs.
I have updated to dbal 3.x, and used `symfony/uid` to handle the UUID generator, but another issues occurred, the migrations does not generated scripts as before, see: https://github.com/doctrine/DoctrineMigrationsBundle/issues/449

The complete example codes are updated, check https://github.com/hantsy/symfony5-sample/tree/master/rest-sample

When add `-vvv`, no any exceptions when running the generate command.

**Reply by SenseException** [reply]
A migration file creation is a diff between the changes in the entity and the database. Code alone doesn't help to find the reason for the empty file (at least for me).

**Reply by hantsy** [reply]
@SenseException Understand now. It is a little different from the flyway(Java).
